src/iarc7_motion/task_command_handler.py

```diff
@@ -0,0 +1,238 @@
+        self._command_implementations = {
+            task_commands.VelocityCommand: self._handle_velocity_command,
+            task_commands.NopCommand: self._handle_nop_command,
+            task_commands.GroundInteractionCommand: self._handle_ground_interaction_command,
+            task_commands.ConfigurePassthroughMode: self._handle_passthrough_mode_command,
+            task_commands.AngleThrottleCommand: self._handle_passthrough_command
+            }
+
+    # takes in new task from HLM Controller
+    # transition is of type TransitionData 
+    def new_task(self, task, transition):
+        self._task = task
+        self._transition = transition
+        self._task_canceled = False
+        self._task_state = task_states.TaskRunning()
+        self._task.send_transition(self._transition)
+
+    # cancels task
+    def cancel_task(self):
+        if self._task is None:
+            rospy.logerr('No task running to cancel.')
+        try:
+            self._task.cancel()
+            self._task_canceled = True
+            self._task = None
+        except Exception as e:
+            rospy.logerr('Exception canceling task')
+            rospy.logerr(str(e))
+            rospy.logerr(traceback.format_exc())
+            rospy.logerr('Task Command Handler aborted task')
+            self._task_state = task_states.TaskAborted()
+            self._task = None
+
+    # returns last command (ground interaction, velocity, etc.)
+    # that task returned
+    def get_last_command(self):
+        return self._last_task_commands
+
+    # returns last twist sent to LLM
+    def get_last_twist(self):
+        return self._last_twist
+    
+    # public method wrapper for getting task state
+    def get_state(self):
+        return self._task_state
+
+    # main function
+    def run(self):
+        task_commands = self._get_task_command()
+        for task_command in task_commands:
+            try:
+                self._command_implementations[type(task_command)](task_command)
+            except (KeyError, TypeError) as e:
+                rospy.logerr("Task requested unimplemented command, noping: %s", type(task_command))
+                self._handle_nop_command(None)
+
+        self._last_task_commands = task_commands
+
+    # gets desired command from running task
+    def _get_task_command(self):
+        if self._task is not None:
+            try:
+                task_request = self._task.get_desired_command()
+            except Exception as e:
+                rospy.logerr('Exception getting tasks preferred velocity')
+                rospy.logerr(str(e))
+                rospy.logerr(traceback.format_exc())
+                rospy.logerr('Task Command Handler aborted task')
+                self._task = None
+                self._task_state = task_states.TaskAborted()
+                return (task_commands.NopCommand(),)
+
+            self._task_state = task_request[0]
```

Comment:
Need to check that the task_request has an element 0 (just wrap this in a try block and catch TypeError and IndexError)